superset-frontend/src/explore/components/controls/MetricControl/AdhocMetricPopoverTrigger.tsx

```diff
@@ -119,6 +124,10 @@ class AdhocMetricPopoverTrigger extends React.PureComponent<
     this.forceUpdate();
   }
 
+  handleDatasetModal(bool: boolean) {
+    this.setState({ showSaveDatasetModal: bool });
```

Comment:
nit, but I wouldn't use `bool` as a variable name for a value, since it reads too closely to a type. I would use something descriptive like `shouldShow`, `showModal`, or even `value` if you want to be very generic.